### PR TITLE
Next major step for rubik's third mission

### LINKED ISSUE #59907: Exodii Mission 2 Expansion: Allow NPCs to help you find the warehouse
### Is your feature request related to a problem? Please describe.

See https://github.com/CleverRaven/Cataclysm-DDA/issues/59900

The Exodii missions haven't yet been implemented, and I am worried that my life commitments will keep them out of my reach for a long time. I am posting my plan for their missions in the hopes that someone will help me implement parts of them. If that happens, I would be more than happy to contribute with dialogue and editing and some of the components as I have time.

Of course I may get to this myself, in which case having it written up will keep me from forgetting what I wanted.

See [Mission 2: Find the lost warehouse](https://github.com/CleverRaven/Cataclysm-DDA/issues/59905): this issue expands the mission so that you can ask NPCs from other factions for advice, making the mission much easier and also more interesting.

### Solution you would like.

Follow up refinements to [Mission 2: Find the Lost Warehouse](https://github.com/CleverRaven/Cataclysm-DDA/issues/59905) to be added after the mission is working in base form.

#### Changes to mission offer dialogue
- Rubik suggests you can ask around amongst other scavengers and travellers for clues, they don't mind if word gets out that you're looking for it.
- If you have previously made contact with the Hub, and Rubik knows that, they might slightly hint that you shouldn't go chatting with those guys, but they won't oversell it because they don't want to give you ideas.

#### Talking to other NPCs
You can ask several NPCs for help to find the warehouse. Possible NPCs to ask include: 
- Free Merchant Teamster
- random Free Merchant roadside encounters
- the Arsonist
- one of the Crafters
- one of the Hub mercenaries
- if possible, any randomly spawned NPC might have a chance of discussing this with you.
- Other NPCs in other factions that suit the 'traveller who might know' could be included as well.

You won't always get the right answer but eventually you are guaranteed to:
- The first NPC you ask has a 1/3 chance of having no idea what you're looking for, a 1/3 chance of giving you directions to a different location, and a 1/3 chance of giving you correct directions.
- If you haven't yet received correct directions, the second NPC you ask has a 1/6 chance of having no idea, a 2/6 chance of giving you incorrect directions, and a 3/6 chance of giving you correct directions.
- If you haven't yet received correct directions, the third NPC you ask will definitely give you correct directions.
- Once you have received correct directions, NPCs have a 50/50 chance of either not knowing or giving you incorrect directions

Possible incorrect directions: You may be given directions to an infested LMOE shelter, a different exodii warehouse, a mi-go tower, a FEMA camp, or any of several low-value non-city map specials.

#### Hinting conflict with Hub
- If you tell the hub01 intercom more about what you are looking for, they will ask you some more questions about it. You have the option of sharing more information. Their reaction will depend a bit on how much they trust you already.
- Actual conflict with the Hub will be added in another refinement Issue to this one, just getting the NPC dialogue and directions mechanic in is sufficient for now.

### Describe alternatives you have considered.

I'm not completely sure if the "wrong directions" mechanic is necessary, but I think it is *fun*. There are other ways it could be implemented. For example we could just have a counter of "how many times have you asked for directions without getting the right answer" and if it is 2 or more, you always get the right answer - no change in likelihood on the first two answers.

Rather than giving you directions to a wide range of maps the wrong directions instead could always lead somewhere useless, like a pile of wreckage in the woods or something.

### Additional context

n/a

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
Content "Finish the basic part of rubik's 'find the warehouse' mission, allowing you to ask around for directions; success is not solely RNG"

#### Purpose of change

Closes #59907

SariusSkelrets added the first part of a major exodii mission in #67695, but in this stub version the mission is a very boring 'find this overmap special' mission. It was left unfinished for a while because, well, busy life, volunteer, etc etc. (this is not the fault of sariusskelrets but of me)

See https://github.com/CleverRaven/Cataclysm-DDA/issues/59905 for a more full list of what I'd like to do with this mission in the long run. I hope to make it into the first chunk where you can decide if you want to help the exodii or sell them out 

#### Describe the solution
The warehouse rubik asks you to find is now globally unique and quite rare, it's gonna be hard to stumble on it. You *can*, I hope, but it is hard. However, now you can ask various NPCs around the world if they've seen it. You start with a 1/6 chance of getting the correct directions, but every time you ask a new person, the chance of getting the right directions increases to 1/5, then 1/4, then 1/3, then 1/2, then a guarantee if you have still lost the lottery.

If they don't know where it is, chances are they'll just admit it, but there's 1 in 4 chance they'll send you somewhere else by accident.

Aside from Smokes and the free merchant teamster, you can ask a bunch of random NPCs around the world:
- the scrap trader
- the lumbermill logger
- the random encounters with john bailey, the refugee caravan, or the camper van trader

I'm open to other candidates, but I'd like to avoid anyone else in the refugee center so that it's not the one-stop shop, and I'd like to avoid anyone in the hub for story reasons

I have also cleaned up rubik's dialogue trees a bit because when going through them I spotted some errors and some parts where they were needlessly frustrating to navigate. I confess I'm a little annoyed nobody ever bug reported these >.>

#### Describe alternatives you've considered

Aside from mildly different implementation I can't think of another way to make the quest more interesting. I plan to add the option to quite a few people around the world so that one way or another you can definitely find someone who's seen it.

After this PR, the next step is to give the player the option to sell the location of the warehouse to the Hub before reporting it back to Rubik. I was thinking of adding it this PR, but I have now got more pressing priorities

#### Testing
It works!
image

I asked two people and got two possible sites to investigate, which is a neat outcome I hadn't actually anticipated.
image

Smokes, you dick, this is not a concrete building with a metal spire sticking out the top.
image

The teamster did better but this still ain't it chief.
image

Took a few rolls o' the dice but we made it in the end.
image

I am shocked to say this all worked on the first test. It definitely made the mission better. I need to clean up rubik's dialogue linkages a lot.

#### Additional context

I can't remember if this method will force generation of it if the special doesn't spawn. It should.

### CODE REVIEW

**Comment by kevingranade** [comment]
We have code for radio triangulation, Rubik could give you a locator device.

**Comment by I-am-Erk** [comment]
> We have code for radio triangulation, Rubik could give you a locator device.

The idea is that they can't do this because the transceiver is down, you're going out to reactivate it for them. As a local guide you're better suited than their quads because you can ask around and see if others have seen it.

**Comment by SariusSkelrets** [comment]
Thank you for finishing the mission. I was incredibly busy these last months (and still am)
It's great to see the Exodii and their content being developped more

File: data/json/npcs/exodii/common_talk.json [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+          { "case": 3, "effect": { "assign_mission": "mission_exodii_wh_bad_directions3" } }
+        ]
+      }
+    },
+    "responses": [ { "text": "Thanks.", "topic": "TALK_DONE" } ]
+  },
+  {
+    "id": "mission_exodii_wh_directions",
+    "type": "mission_definition",
+    "name": { "str": "Possible directions to exodii warehouse" },
```
Comment (Qrox) [inline data/json/npcs/exodii/common_talk.json:69]:
```suggestion
    "name": { "str": "Possible directions to Exodii warehouse" },
```

File: data/json/npcs/exodii/common_talk.json [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+      "inquire": ".",
+      "success": ".",
+      "success_lie": ".",
+      "failure": "."
+    }
+  },
+  {
+    "id": "mission_exodii_wh_bad_directions1",
+    "type": "mission_definition",
+    "name": { "str": "Possible directions to exodii warehouse" },
```
Comment (Qrox) [inline data/json/npcs/exodii/common_talk.json:98]:
```suggestion
    "name": { "str": "Possible directions to Exodii warehouse" },
```

File: data/json/npcs/exodii/common_talk.json [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+      "inquire": ".",
+      "success": ".",
+      "success_lie": ".",
+      "failure": "."
+    }
+  },
+  {
+    "id": "mission_exodii_wh_bad_directions2",
+    "type": "mission_definition",
+    "name": { "str": "Possible directions to exodii warehouse" },
```
Comment (Qrox) [inline data/json/npcs/exodii/common_talk.json:127]:
```suggestion
    "name": { "str": "Possible directions to Exodii warehouse" },
```

File: data/json/npcs/exodii/common_talk.json [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+      "inquire": ".",
+      "success": ".",
+      "success_lie": ".",
+      "failure": "."
+    }
+  },
+  {
+    "id": "mission_exodii_wh_bad_directions3",
+    "type": "mission_definition",
+    "name": { "str": "Possible directions to exodii warehouse" },
```
Comment (Qrox) [inline data/json/npcs/exodii/common_talk.json:156]:
```suggestion
    "name": { "str": "Possible directions to Exodii warehouse" },
```

**Review by Qrox — COMMENTED** [review]
Capitalize Exodii, to be consistent with rest of the in-game text.

**Pull request merged**